docs/how-tos/how-to-reset-state.mdx

```diff
@@ -34,3 +34,34 @@ const reset = () => {
   state.obj = _.cloneDeep(initialObj)
 }
 ```
+
+> [!NOTE]
+> Using `structuredClone()`
+
+In 2022, there was a new global function added called `structuredClone` that is widely available in most modern browsers. While this function and `_.cloneDeep` from lodash seem to do the same thing, their implementations are quite different.
+
+In valtio v1, `proxy` cloned the object used for inital state under the hood. Because the original object isn't used, `structuredClone` works just fine as a drop in replacement for `_.cloneDeep`.
```

Comment:
Is this only the issue with `structuredClone`? Can it be the same with `_.cloneDeep`?

Reply:
It doesn't seem to. I haven't tested very thoroughly but from what I've tested so far, it seems to work ok, even modifying the structure of the object doesn't really seem to affect it too much. This example doesn't seem to have any issues

```js
import { proxy, useSnapshot } from 'valtio'
import { cloneDeep } from 'lodash'

const initialState = {
  foo: {
    bar: {
      bat: 'bat initial'
    }
  }
}
const state = proxy({ obj: initialState })
const reset = () => {
  state.obj = cloneDeep(initialState)
}

function App() {
  const snap = useSnapshot(state)

  return (
    <div>
      <p>snap.foo.bar.bat: {snap.obj.foo.bar.bat}</p>
      <button onClick={() => state.obj = { foo: { bar: { asdf: 'bat2', ball: 'ball' } }}}>click</button>
      <button onClick={() => state.obj = { foo: { bar: 'bar2'}}}>click</button>
      <button onClick={() => reset()}>reset</button>
      <pre>{JSON.stringify(snap.obj, null, 2)}</pre>
    </div>
  )
}

export default App

```

Reply:
Outputs as expected:

Initial state
```
snap.foo.bar.bat: bat initial

{
  "foo": {
    "bar": {
      "bat": "bat initial"
    }
  }
}
```

First button click
```
snap.foo.bar.bat:

{
  "foo": {
    "bar": {
      "asdf": "bat2",
      "ball": "ball"
    }
  }
}
```

Second button click
```
snap.foo.bar.bat:

{
  "foo": {
    "bar": "bar2"
  }
}
```

Reset button click
```
snap.foo.bar.bat: bat initial

{
  "foo": {
    "bar": {
      "bat": "bat initial"
    }
  }
}
```

Reply:
Can you try this?

```jsx
<button onClick={() => { state.obj.foo.bar.bat = 'bat changed' }}>click</button>
```

Reply:
yeah that does mess it up. I'll update the PR once I get a chance.

Reply:
I updated the docs. Let me know if you want anything reworded.